compose/service.py

```diff
@@ -678,10 +678,21 @@ def labels(self, one_off=False):
         ]
 
     def can_be_scaled(self):
-        for port in self.options.get('ports', []):
-            if ':' in str(port):
+        # TODO: this function should ideally move to docker.utils.ports
+        # get rid of it once its accepted into docker-py
+        def has_external_port(external_binding):
+            if external_binding is None:
```

Comment:
If it's going to be a public function in docker-py, it needs to be able to handle all valid port bindings, not just the subset created by `split_port()`.